core/src/main/scala/org/apache/spark/SparkContext.scala

```diff
@@ -689,6 +689,64 @@ class SparkContext(config: SparkConf) extends Logging with ExecutorAllocationCli
     new ParallelCollectionRDD[T](this, seq, numSlices, Map[Int, Seq[String]]())
   }
 
+  /**
+   * Creates a new RDD[Long] containing elements from `start` to `end`(exclusive), increased by
+   * `step` every element.
+   *
+   * @note if we need to cache this RDD, we should make sure each partition contains no more than
+   *       2 billion element.
+   *
+   * @param start the start value.
+   * @param end the end value.
+   * @param step the incremental step
+   * @param numSlices the partition number of the new RDD.
+   * @return
+   */
+  def range(
+      start: Long,
+      end: Long,
+      step: Long = 1,
+      numSlices: Int = defaultParallelism): RDD[Long] = withScope {
+    assertNotStopped()
+    if (step == 0) {
+      // when step is 0, range will run infinite
+      throw new IllegalArgumentException("`step` cannot be 0")
+    }
+    val length =
+      if ((end - start) % step == 0) {
```

Comment:
This is really an issue.
The length could be overflow also. Even if the `length` does not overflow, the calculation of `partitionStart`/`partitionEnd` could also lead to that. I am trying to figure out another way to calculate this, since the `length` is used for this calculation.